### PR TITLE
custom databaker function - like_observations

### PR DESCRIPTION
**Pull request by osamede20**
https://github.com/GSS-Cogs/de-helpers/issues/1

__init_py updated to include like_observations library
Observation function - observation.py created to pick out observations in transforms
test_observation.py - unit tests created to test the observation function

### CODE REVIEW

**Comment by mikeAdamss** [comment]
@osamede20 - Some nice hacking at a complex problem here and I can see where you're going with this, but I think is probably too ambitious for a helper.

Think of: [https://github.com/GSS-Cogs/de-helpers/blob/785296a6232b4800da174bd6b2de855c6560ea36/dehelpers/filters/observation.py#L18](https://github.com/GSS-Cogs/de-helpers/blob/785296a6232b4800da174bd6b2de855c6560ea36/dehelpers/filters/observation.py#L18) , it's always gonna be possible to have actual observations that look like years, 1998, 2005 etc are also valid observable values, we can't automate that away.

We also need to avoid imports, this branch is failing tests because it doesn't have gssutils and databaker installed and thats very much intended. We need to be very careful to avoid the scenario where the transforms use gssutils, databaker and the helpers but the helpers also use gssutils and databaker, its not quite a circular dependency but it's close enough that it'll cause problems (for example: you couldn't really release either gssutils or databaker without also releasing this library). 

Basically, a helpers needs to be a small simple thing.

If you want to do a like_observation filter then just write something that return True if you can `float(cell.value)` without an error. If you want to go beyond that have it accept data markers as well, then look at something along the lines of:
```
like_observation(allow_markers=["*", "-"])
```

But leave it at that, it's fine to leave the DE to figure our what's an observations vs whats a year and to use some sense when applying filters, if you try and automate that for all possible transforms you'll end up with a staggeringly complex thing (I suspect you'd need some form of AI:smile:).

So points for ambition, but simplify!

**Comment by osamede20** [comment]
@mikeAdamss 
Thanks for taking the time to carefully explain and give the feedback.

I had some thoughts within that I was probably making the task more difficult than it should be, Yes, I should have been more careful and given more thoughts to not having the helpers use gssutils and databaker. 

Yes, also, I noticed that the branch is failing tests because it doesn't have gssutils and databaker installed. I wanted to wait for the PR review before asking questions relating to the failing tests. But you have explained the problems the dependency between the helpers on one hand and gssutils and databaker on another can cause. 

Like you advised, I will write a simple filter for observations but also like you mentioned, the only issue is that it might pick up years. But DEs can sort that out later.

Again, thanks for the feedback

**Comment by mikeAdamss** [comment]
closing, duplicate
